### ISSUE TITLE
Buttons are invisible if primary color is white or black

### ISSUE THREAD

**Issue by tamasg** [squidfunk/mkdocs-material#4571]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

Button is wrong if the palette is white.

### Expected behaviour

For example, if it is red.

https://user-images.githubusercontent.com/5910815/198908323-32216103-69d6-4356-b30c-34bb103d57d3.mp4

### Actual behaviour

https://user-images.githubusercontent.com/5910815/198908339-a40a5b06-32fd-4863-9fe5-57b4b5595134.mp4

Cookie

https://user-images.githubusercontent.com/5910815/198908355-1c0b8880-d12f-4ab8-8683-78fcf03bbf54.mp4

### Steps to reproduce

Just add buttons:

```markdown 
[Subscribe to our newsletter](#){ .md-button }

[Subscribe to our newsletter](#){ .md-button .md-button--primary }
```

### Package versions

mkdocs: 1.4.1
mkdocs-material: 8.5.7+insiders.4.26.1

### Configuration

```yaml
theme:
  name: material
  palette:
    scheme: default
    primary: white
```

### System information

- Operating system: Windows 10
- Browser: Chrome

**Comment by squidfunk** [comment]
Thanks for reporting. Fixed in 3c5f94f59.

**Comment by tamasg** [comment]
Thanks for the quick fix!

**Comment by squidfunk** [comment]
Released as part of 8.5.8.
